### PR TITLE
Fix structure, table of contents errors, minor code line errors

### PR DESCRIPTION
**Pull request by fhajredini**

### CODE REVIEW

**Review by sashko — CHANGES_REQUESTED** [review]
title

File: docs/articles/architecture/arch-intro.rst [hunk 1]
```diff
@@ -24,6 +24,7 @@ Reader's guide
 - The :ref:`arch-template` is a base document used for creating 
   component and subsystem architecture documentation.
 - The :ref:`arch-glossary` clarifies terms and abbreviations.
+- The :ref:`arch-layers` presents the layers used in pelux.
```
Comment (sashko) [inline docs/articles/architecture/arch-intro.rst]: not layer?

**Review by sashko — CHANGES_REQUESTED** [review]

File: docs/articles/architecture/arch-intro.rst (hunk 1 above)
Comment (fhajredini) [reply docs/articles/architecture/arch-intro.rst]: Afaik, layers.

**Review by FlorentRevest — APPROVED** [review]

**Review by sashko — APPROVED** [review]

File: docs/articles/architecture/arch-layers.rst [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 .. _arch-layer:
 
-Layers maintained by the PELUX project:
+Layers maintained by the PELUX project
 =======================================
```
Comment (TSAnsari) [inline docs/articles/architecture/arch-layers.rst]: Align '=' with 't' in 'project' on above line

**Review by TSAnsari — APPROVED** [review]
Just fix the comment and I am ok.

**Pull request merged**